### DISCUSSION TITLE
How the API routes should be consumed?

### DISCUSSION THREAD

**Discussion by borsemayur2** [alephjs/aleph.js#187] [Q&A] [answered]
Is it okay to use axios/fetch to consume the API routes in pages?

**Comment by shadowtime2000** [comment, marked as answer]
Really depends. If it is data that is supposed to be available during SSR/SSG (`useDeno`) then you shouldn't be doing that. You should instead then be writing the code of the API route inside the page. If it is something that is specifically not meant to be prerendered, then yeah it's pretty much fine.

**Reply by borsemayur2** [reply]
Okay. I think I need a better understanding of SSR and SSG. I'm using API routes for CRUD actions. Thanks

**Reply by borsemayur2** [reply]
It means, using axios to consume API routes makes it **Client Side Rendered** page. Right?

**Reply by shadowtime2000** [reply]
@borsemayur2 So if your using API routes for CRUD actions then its fine to use axios/fetch
